app/src/ui/dropdown-select-button.tsx

```diff
@@ -91,132 +102,168 @@ export class DropdownSelectButton<
     this.setState({ showButtonOptions: !this.state.showButtonOptions })
   }
 
+  private onDropdownButtonKeyDown = (
+    event: React.KeyboardEvent<HTMLButtonElement>
+  ) => {
+    const { key } = event
+    let flag = false
+
+    switch (key) {
+      case ' ':
+      case 'Enter':
+      case 'ArrowDown':
+      case 'Down':
+        this.setState({
+          selectedOption: this.props.options.at(0) ?? null,
+          showButtonOptions: true,
+        })
+        flag = true
+        break
+
+      case 'Esc':
+      case 'Escape':
+        this.dropdownButtonRef?.focus()
+        this.setState({ showButtonOptions: false })
+        flag = true
+        break
+
+      case 'Up':
+      case 'ArrowUp':
+        this.setState({
+          selectedOption: this.props.options.at(-1) ?? null,
+          showButtonOptions: true,
+        })
+        flag = true
+        break
+
+      default:
+        break
+    }
+
+    if (flag) {
+      event.stopPropagation()
+      event.preventDefault()
+    }
+  }
+
   private onInvokeButtonRef = (buttonRef: HTMLButtonElement | null) => {
     this.invokeButtonRef = buttonRef
   }
 
+  private onDropdownButtonRef = (buttonRef: HTMLButtonElement | null) => {
+    this.dropdownButtonRef = buttonRef
+  }
+
   private onOptionsContainerRef = (ref: HTMLDivElement | null) => {
     this.optionsContainerRef = ref
   }
 
-  private renderSelectedIcon(option: IDropdownSelectButtonOption<T>) {
-    const { selectedOption } = this.state
-    if (selectedOption === null || option.value !== selectedOption.value) {
+  private renderOption = (item: MenuItem) => {
+    const option = this.props.options.find(o => o.value === item.id)
+    if (!option) {
       return
     }
 
     return (
-      <Octicon
-        className="selected-option-indicator"
-        symbol={OcticonSymbol.check}
-      />
-    )
-  }
-
-  private renderOption = (o: IDropdownSelectButtonOption<T>) => {
-    return (
-      <Button key={o.value} onClick={this.onSelectionChange(o)}>
-        {this.renderSelectedIcon(o)}
-        <div className="option-title">{o.label}</div>
-        <div className="option-description">{o.description}</div>
-      </Button>
+      <>
+        <div className="option-title">{option.label}</div>
+        <div className="option-description">{option.description}</div>
+      </>
     )
   }
 
   private renderSplitButtonOptions() {
-    if (!this.state.showButtonOptions) {
+    const {
+      showButtonOptions,
+      checkedOption,
+      selectedOption,
+      optionsPositionBottom: bottom,
+    } = this.state
+    if (!showButtonOptions) {
       return
     }
 
     const { options } = this.props
-    const { optionsPositionBottom: bottom } = this.state
+
+    const items: ReadonlyArray<MenuItem> = options.map(o => ({
+      type: 'checkbox',
+      id: o.value,
+      enabled: true,
+      visible: true,
+      label: o.label as string, // TODO: Accept string - use render for jsx
```

Comment:
Woops. I forgot to see if the label in the dropdown option interface had a use case as a jsx or if it could be just a string.. will follow this up..

Reply:
Refactored dropdown option to use only a string for the `label` property as well as renaming the `value` property to be `id` to be inline with the `MenuItem` type. This resulted in the removal of the generic typing.